### PR TITLE
[GLUTEN-2604][VL] Fix github dependabot alert for Spark version

### PR DESCRIPTION
**Pull request by xieqi**
## What changes were proposed in this pull request?

Fix dependabot alert

```
Apache Spark vulnerable to Improper Privilege Management

Upgrade org.apache.spark:spark-core_2.12 to fix [1 Dependabot alert](https://github.com/oap-project/gluten/security/dependabot?q=is%3Aopen+package%3Aorg.apache.spark%3Aspark-core_2.12+manifest%3Atools%2Fgluten-it%2Fpom.xml+has%3Apatch) in [tools/gluten-it/pom.xml](https://github.com/oap-project/gluten/blob/-/tools/gluten-it/pom.xml)
Upgrade org.apache.spark:spark-core_2.12 to version 3.4.0 or later.

In Apache Spark versions prior to 3.4.0, applications using spark-submit can specify a proxy-user to run as, limiting privileges. The application can execute code with the privileges of the submitting user, however, by providing malicious configuration-related classes on the classpath. This affects architectures relying on proxy-user, for example those using Apache Livy to manage submitted applications.

Update to Apache Spark 3.4.0 or later, and ensure that spark.submit.proxyUser.allowCustomClasspathInClusterMode is set to its
default of "false", and is not overridden by submitted applications.

```
## How was this patch tested?

Use exist UTs

### CODE REVIEW

File: tools/gluten-it/README.md [hunk 1]
```diff
@@ -14,14 +14,17 @@ https://github.com/oap-project/gluten
 
 See official Gluten build guidance https://github.com/oap-project/gluten#how-to-use-gluten
 
-### 2. Install and run gluten-it
+### 2. Install and run gluten-it with Spark version
+
+> Note: Support Spark 3.2 and Spark 3.3 only
```
Comment (winningsix) [inline tools/gluten-it/README.md:19]: Seems we don't have a place to maintain supported Spark versions, OS and HW. Is it good to centralize it in README?

File: tools/gluten-it/README.md [hunk 2]
```diff
@@ -14,14 +14,17 @@ https://github.com/oap-project/gluten
...
 
 ```sh
 cd gluten/tools/gluten-it
-mvn clean package
+mvn clean package -Pspark-3.2
```
Comment (winningsix) [inline tools/gluten-it/README.md:23]: mvn clean package -P{Spark-Version}

File: tools/gluten-it/README.md [hunk 3]
```diff
@@ -14,14 +14,17 @@ https://github.com/oap-project/gluten
...
 sbin/gluten-it.sh
 ```
 
+
```
Comment (winningsix) [inline tools/gluten-it/README.md:27]: Trivial note: undesired line break?

File: tools/gluten-it/README.md (hunk 1 above)
Comment (xieqi) [reply tools/gluten-it/README.md]: ok, will add in another doc update PR together with how to user header checker and how to download pre-build jar.

**Comment by zhouyuan** [comment]
@zhztheplayer this patch tries to upgrade the spark jars used in gluten-it to 3.4 to fix the security issue

-yuan

**Review by zhouyuan — APPROVED** [review]

**Review by zhztheplayer — APPROVED** [review]
Can we attach the alert text to the PR description?

**Comment by xieqi** [comment]
Apache Spark vulnerable to Improper Privilege Management

Upgrade org.apache.spark:spark-core_2.12 to fix [1 Dependabot alert](https://github.com/oap-project/gluten/security/dependabot?q=is%3Aopen+package%3Aorg.apache.spark%3Aspark-core_2.12+manifest%3Atools%2Fgluten-it%2Fpom.xml+has%3Apatch) in [tools/gluten-it/pom.xml](https://github.com/oap-project/gluten/blob/-/tools/gluten-it/pom.xml)
Upgrade org.apache.spark:spark-core_2.12 to version 3.4.0 or later.

In Apache Spark versions prior to 3.4.0, applications using spark-submit can specify a proxy-user to run as, limiting privileges. The application can execute code with the privileges of the submitting user, however, by providing malicious configuration-related classes on the classpath. This affects architectures relying on proxy-user, for example those using Apache Livy to manage submitted applications.

Update to Apache Spark 3.4.0 or later, and ensure that spark.submit.proxyUser.allowCustomClasspathInClusterMode is set to its
default of "false", and is not overridden by submitted applications.

File: tools/gluten-it/pom.xml [hunk 4]
```diff
@@ -143,5 +140,33 @@
         </dependency>
       </dependencies>
     </profile>
+    <profile>
+      <id>spark-3.4</id>
```
Comment (jinchengchenghh) [inline tools/gluten-it/pom.xml:144]: Does it support spark-3.4?

**Comment by zhouyuan** [comment]
@zhztheplayer PR desc updated.

**Pull request merged**